addons/sale_project/models/project_task.py

```diff
@@ -55,7 +55,7 @@ def _group_expand_sales_order(self, sales_orders, domain):
             return search_on_comodel
         return sales_orders
 
-    @api.depends('sale_line_id', 'project_id', 'allow_billable')
+    @api.depends('sale_line_id', 'project_id', 'allow_billable', 'project_id.reinvoiced_sale_order_id', 'project_id.sale_order_id', 'project_id.reinvoiced_sale_order_id.state')
```

Comment:
```suggestion
    @api.depends('sale_line_id', 'project_id', 'allow_billable', 'project_id.reinvoiced_sale_order_id', 'project_id.reinvoiced_sale_order_id.state')
```

It seems that project_id.sale_order_id is not required